### PR TITLE
Trace hash_graph and fuse_optimizers with Scalopus

### PR DESCRIPTION
**Pull request by efernandez**
This adds tracepoints to the `HashGraph` class and the `fuse_optimizers` using [scalopus](https://github.com/iwanders/scalopus).

This relies on the private ROS package `cpr_scalopus`. That's why this PR is against `cpr-devel` and in our fork.

Example:
t2_fuse_rl_scalopus

### CODE REVIEW

File: fuse_optimizers/src/batch_optimizer.cpp [hunk 1]
```diff
@@ -184,16 +188,21 @@ void BatchOptimizer::optimizationLoop()
...
-    graph_->optimize();
-    // Make a copy of the graph to share
-    fuse_core::Graph::ConstSharedPtr const_graph = graph_->clone();
-    // Optimization is complete. Notify all the things about the graph changes.
-    notify(const_transaction, const_graph);
-    // Clear the request flag now that this optimization cycle is complete
-    optimization_request_ = false;
+
+    {
+      TRACE_SCOPE_RAII("Optimize.");
```
Comment (iwanders) [inline fuse_optimizers/src/batch_optimizer.cpp]: I would remove the dot at the end, just make it `Optimize`, there's one down below as well. But it's a matter of taste.

**Comment by iwanders** [comment]
Looks good @efernandez. :+1:

**Review by iwanders — APPROVED** [review]

File: fuse_optimizers/src/batch_optimizer.cpp (hunk 1 above)
Comment (efernandez) [reply fuse_optimizers/src/batch_optimizer.cpp]: I saw some code doing that, but I actually prefer it w/o the trailing dot. :smiley:

Comment (efernandez) [reply fuse_optimizers/src/batch_optimizer.cpp]: Changed

**Pull request merged**